### ISSUE TITLE
Use tent

### ISSUE THREAD

**Issue by thatmacguy** [michaelchadwick/gemwarrior#21]
I was horribly mangled by an amethystle,  I attempted to rest in my tent to regain some health:

```
LV: 1][XP:  5][ROX:  6] [HP: 21/30 ][STM:20/20] [Larianele @ Forest (Southwest)]
 GW> use tent
tent can be used 4 more time(s).
no implicit conversion from nil to integer
######################################################################
Gem Warrior played for 4 minutes, 8 seconds, and 201 milliseconds
----------------------------------------------------------------------
Player killed 1 monster(s)
       picked up 4 item(s)
       traveled 8 time(s)
       rested 0 time(s)
######################################################################
```

**Comment by michaelchadwick** [comment]
There must be more to this bug, as I've tried to recreate it several different ways and cannot. Can you retrace roughly the steps you took before using the tent?

**Comment by thatmacguy** [comment]
Here is the entire game play:

```
========================================================================
Welcome to *Jool*, where randomized fortune is just as likely as mayhem.
========================================================================
* Remember: bains and splats are the key to civility *

[ Home ]
The little, unimportant, decrepit hut that you live in.
>> Curious object(s): bed, letter, stone, tent
>> Paths: north, east, west

[LV: 1][XP:  0][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Home]
 GW> t stone
Added stone to your increasing collection of bits of tid.

[LV: 1][XP:  0][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Home]
 GW> e stone
The stone has taken charge, and been equipped.

[LV: 1][XP:  0][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Home]
 GW> l letter
A single page of thin paper, folded at the middle, with some excellent
penmanship impressed upon it.

[LV: 1][XP:  0][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Home]
 GW> t letter
Added letter to your increasing collection of bits of tid.

[LV: 1][XP:  0][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Home]
 GW> l letter
A single page of thin paper, folded at the middle, with some excellent
penmanship impressed upon it.

[LV: 1][XP:  0][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Home]
 GW> e letter
That cannot be equipped as a weapon.

[LV: 1][XP:  0][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Home]
 GW> ?
That is not something the game yet understands.

[LV: 1][XP:  0][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Home]
 GW> help
=========================================================
 character, c  -- Display character information
 inventory, i  -- Look in your inventory
 rest     , r  -- Take a load off and regain HP
 look     , l  -- Look around your current location
 take     , t  -- Take item
 use      , u  -- Use item (in inventory or environment)
 drop     , d  -- Drop item
 equip    , e  -- Equip item
 unequip  , ue -- Unequip item
 go       , g  -- Go in a direction
 attack   , a  -- Attack a monster
 change   , ch -- Change something
 help     , h  -- This help menu
 quit     , q  -- Quit w/ confirmation (also exit/x)
 quit!    , qq -- Quit w/o confirmation (also exit!/xx)
=========================================================

[LV: 1][XP:  0][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Home]
 GW> use letter
The words of the queen echo in your head as you read the royal note sent to you again:

  DEAR LARIANELE,

    OH, MY! JOOL IS IN TROUBLE! THE EVIL WIZARD/SORCEROR/CONJUROR/ROCKSMITH/WILY 
    EMERALD HAS ABSCONDED WITH OUR SHINYTHING(TM)! IT IS VITAL THAT YOU, LARIANELE, 
    GO TO HIS TOWER IN THE SKY IN ORDER TO RETRIEVE IT BEFORE HE DOES SOMETHING
    TERRIBLE WITH IT!

    REMEMBER THAT ONE TIME YOU CAME TO THE CASTLE, TRYING TO SELL STONES YOU
    PILFERED FROM A NEARBY CAVE? REMEMBER HOW I LAUGHED AND TOLD YOU TO LEAVE
    AT ONCE OR I'D HAVE THE ROYAL GUARD TAKE YOUR HEAD OFF? HA!

    WHAT A FOOL I WAS TO CAST SUCH A SPECIAL PERSON OUT, AS A MYSTERIOUS STRANGER
    IN THE NIGHT TOLD ME, BEFORE MYSTERIOUSLY DISAPPEARING, THAT YOU, LARIANELE,
    ARE ACTUALLY THE ONLY ONE WHO CAN SAVE US (FOR SOME REASON, BUT THAT'S
    MYSTERIOUS STRANGERS FOR YOU, RIGHT?)!

    PLEASE, I BEG OF YOU, SAVE JOOL FROM THE POTENTIAL TERROR THAT EMERALD COULD
    POSSIBLY WREAK ON ALL OF US BEFORE IT IS TOO LATE! IF YOU DO, YOU, LARIANELE,
    WILL BE REWARDED HANDSOMELY!

  SINCERELY,
  QUEEN RUBY

[LV: 1][XP:  3][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Home]
 GW> l
[ Home ]
The little, unimportant, decrepit hut that you live in.
>> Curious object(s): bed, tent
>> Paths: north, east, west

[LV: 1][XP:  3][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Home]
 GW> t tent
Added tent to your increasing collection of bits of tid.

[LV: 1][XP:  3][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Home]
 GW> e
You cannot just "equip". You gotta choose something to equip.

[LV: 1][XP:  3][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Home]
 GW> g e
* >>> *
[ Cave (Entrance) ]
A nearby, dank entrance to a cavern, surely filled with stacktites, stonemites,
and rocksites.
>> Paths: east, west

[LV: 1][XP:  3][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Cave (Entrance)]
 GW> gw
That is not something the game yet understands.

[LV: 1][XP:  3][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Cave (Entrance)]
 GW> g w
* <<< *
[ Home ]
The little, unimportant, decrepit hut that you live in.
>> Curious object(s): bed
>> Paths: north, east, west

[LV: 1][XP:  3][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Home]
 GW> g w
* <<< *
[ Forest (Southeast) ]
Trees exist here, in droves.
>> Curious object(s): tree
>> Paths: north, east

[LV: 1][XP:  3][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Forest (Southeast)]
 GW> g n
* ^^^ *
[ Forest (Northeast) ]
You see much foliage.
>> Paths: south, west

[LV: 1][XP:  3][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Forest (Northeast)]
 GW> g w
* <<< *
[ Forest (Northwest) ]
The amount of leaves and branches and trunks you can see boggles the mind.
>> Curious object(s): feather
>> Paths: east, south

[LV: 1][XP:  3][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Forest (Northwest)]
 GW> t feather
Added feather to your increasing collection of bits of tid.

[LV: 1][XP:  3][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Forest (Northwest)]
 GW> g s
* VVV *
[ Forest (Southwest) ]
While you continue to be impressed at the sheer size of this forest, a clearing
appears to open up to the west.
>> Paths: north, west

[LV: 1][XP:  3][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Forest (Southwest)]
 GW> gw
That is not something the game yet understands.

[LV: 1][XP:  3][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Forest (Southwest)]
 GW> g w
* <<< *
[ Pain Quarry (Southeast) ]
Horrible terribleness emanates from this desolate land of unkind misery.
>> Monster(s) abound: diaman
>> Boss(es) abound: Garynetty
>> Paths: north, east, west

[LV: 1][XP:  3][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Pain Quarry (Southeast)]
 GW> l diaman
Crystalline structure in the form of a man, lumbering toward you, with
outstretched, edged pincers.

[LV: 1][XP:  3][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Pain Quarry (Southeast)]
 GW> g e
* >>> *
[ Forest (Southwest) ]
While you continue to be impressed at the sheer size of this forest, a clearing
appears to open up to the west.
>> Monster(s) abound: amethystle
>> Paths: north, west

[LV: 1][XP:  3][ROX:  0] [HP: 30/30 ][STM:20/20] [Larianele @ Forest (Southwest)]
 GW> a amethystle
******************************
You decide to attack amethystle!
amethystle cries out: "You've found yourself in quite the thorny issue!"

LARIANELE    ::  30 HP
AMETHYSTLE   :: ??? HP

What do you do?
[Fight/Attack][Defend][Look][Pass][Run]
f
You attack amethystle with your stone!
> You wound amethystle for 2 point(s)!
amethystle attacks you!
> You are wounded for 4 point(s)!

LARIANELE    ::  26 HP
AMETHYSTLE   :: ??? HP

What do you do?
[Fight/Attack][Defend][Look][Pass][Run]
f
You attack amethystle with your stone!
> You wound amethystle for 1 point(s)!
amethystle attacks you!
> You are wounded for 5 point(s)!

LARIANELE    ::  21 HP
AMETHYSTLE   :: ??? HP

What do you do?
[Fight/Attack][Defend][Look][Pass][Run]
f
You attack amethystle with your stone!
> You wound amethystle for 1 point(s)!
amethystle attacks you!
amethystle misses entirely!

LARIANELE    ::  21 HP
AMETHYSTLE   :: ??? HP

What do you do?
[Fight/Attack][Defend][Look][Pass][Run]
f
You attack amethystle with your stone!
> You wound amethystle for 2 point(s)!
You have defeated amethystle!

You get the following spoils of war:
 XP   : 2
 ROX  : 6
 ITEMS: herb
******************************

[LV: 1][XP:  5][ROX:  6] [HP: 21/30 ][STM:20/20] [Larianele @ Forest (Southwest)]
 GW> l
[ Forest (Southwest) ]
While you continue to be impressed at the sheer size of this forest, a clearing
appears to open up to the west.
>> Paths: north, west

[LV: 1][XP:  5][ROX:  6] [HP: 21/30 ][STM:20/20] [Larianele @ Forest (Southwest)]
 GW> use tent
tent can be used 4 more time(s).
no implicit conversion from nil to integer
######################################################################
Gem Warrior played for 4 minutes, 8 seconds, and 201 milliseconds
----------------------------------------------------------------------
Player killed 1 monster(s)
       picked up 4 item(s)
       traveled 8 time(s)
       rested 0 time(s)
######################################################################
```

**Comment by michaelchadwick** [comment]
Did you do a `gem update`? I feel like you're not using the latest version.
Running a `gemwarrior -v` should be 0.9.16.

**Comment by thatmacguy** [comment]
hmmm, i did run gem update this morning...

```
Updating gemwarrior
Fetching: gemwarrior-0.9.15.gem (100%)
Successfully installed gemwarrior-0.9.15
Parsing documentation for gemwarrior-0.9.15
Installing ri documentation for gemwarrior-0.9.15
Installing darkfish documentation for gemwarrior-0.9.15
```

well, there's yer problem...

**Comment by michaelchadwick** [comment]
I'm a little confused: does this mean that the problem is solved?

**Comment by thatmacguy** [comment]
let call this a duplicate of #30 and #32

**Comment by michaelchadwick** [comment]
So, you're still experiencing this, even on the latest version? Can you
reproduce it at will?

**Comment by michaelchadwick** [comment]
This is (finally?) fixed in [0.10.4](https://github.com/michaelchadwick/gemwarrior/releases/tag/v0.10.4).
